components/mediation/data-publishers/org.wso2.micro.integrator.observability/src/main/java/org/wso2/micro/integrator/observability/handler/metrichandler/MetricReporter.java

```diff
@@ -0,0 +1,138 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.micro.integrator.observability.handler.metrichandler;
+
+import java.util.Map;
+
+/**
+ * The Metric Reporter allows one to access the relevant Metric Reporter implementation.
+ */
+public interface MetricReporter {
+
+    /**
+     * Creates a metric of the provided type with the given metric name,
+     * the metric description and the set of properties that can be used
+     * to identify a metric uniquely and for metric aggregation, metric filtering.
+     *
+     * @param serviceType The service type (Proxy Service, API, Inbound endpoint )
+     *                    for which the metric is going to be instrumented.
+     * @param type        The type of metric (e.g.Counter) that is used in instrumentation.
+     * @param metricName  Metric name (The proxy/api/inbound endpoint name)
+     * @param metricHelp  Metric description
+     * @param properties  Metric labels used to uniquely identify a metric.
+     */
+    void createMetrics(String serviceType, String type, String metricName, String metricHelp,
+                       String[] properties);
+
+    /**
+     * This is called in the handleInit() method of the AbstractExtendedSynapseHandler.
+     * There by the metrics created in the createMetric() method are invoked.
+     */
+    void initMetrics();
+
+    /**
+     * Creates a metric of the provided type when a request invocation fails
+     * with the given metric name, the metric description and the set of
+     * properties that can be used to identify a metric uniquely and for
+     * metric aggregation, metric filtering.
+     *
+     * @param serviceType Service type
+     * @param type        Metric type
+     * @param metricHelp  Metric description
+     * @param properties  Metric labels
+     */
+    void initErrorMetrics(String serviceType, String type, String metricName, String metricHelp, String[] properties);
+
+    /**
+     * Increment the metric value when a request is received/server/service is deployed.
+     *
+     * @param metricName Metric name
+     * @param properties Metric labels where the Metric and the labels are defined as key-value pairs of the properties
+     *                   Map
+     */
+    void incrementCount(String metricName, String[] properties);
+
+    /**
+     * Decrement the metric value when a request is received/server/service is un-deployed.
+     *
+     * @param metricName Metric name
+     * @param properties Metric and the labels as key-value pairs of the properties Map
+     */
+    void decrementCount(String metricName, Map<String, String> properties);
```

Comment:
Let's change this Map to String[] as well to unify the implementation.

Reply:
Fixed